common/openstack/src/main/java/org/jclouds/openstack/handlers/RetryOnRenew.java

```diff
@@ -116,7 +115,11 @@ public boolean shouldRetryRequest(HttpCommand command, HttpResponse response) {
          return retry;
 
       } finally {
-         releasePayload(response);
+         // If the request is failed and is not going to be retried, the
+         // ErrorHandler will be invoked and it might need to read the payload.
+         // For some kind of payload sources, such as the OkHttp Source, if the
+         // payload is released, the upcoming operations will fail.
+         closeClientButKeepContentStream(response);
```

Comment:
Is the retry handler doing something with the payload? If not, I'd just remove the finally block to avoid buffering it in memory unnecessarily.